File: my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

"""Este node cria um nó simples que registra "Hello world" ao iniciar 
e depois registra "Hello" seguido de um contador a cada segundo.
"""
class MyNode(Node):

    def __init__(self):
        super().__init__("py_test") # criar um nó chamado "py_test"

        self.counter_ = 0

        self.get_logger().info("Hello world") # registrar uma mensagem de log
        
        # criar um timer que chama timer_callback a cada 1 segundo
        self.create_timer(1.0, self.timer_callback) 

    # função chamada pelo timer a cada segundo
    def timer_callback(self): 
        self.get_logger().info("Hello" + str(self.counter_))
        self.counter_ += 1


def main(args=None):
    rclpy.init(args=args) # inicializar o rclpy
    node = MyNode()       # instanciar o nó
    rclpy.spin(node)      # manter o nó ativo para processar callbacks(até que se aperte Ctrl+C)
    rclpy.shutdown()      # finalizar o rclpy
#-----------------------------------------------------------------------------------------------
if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp" 

class MyNode : public rclcpp::Node // define a classe MyNode que herda de rclcpp::Node
{

public:
    MyNode() : Node("cpp_test"), counter_(0)
    {
        RCLCPP_INFO(this->get_logger(), "Hello World"); // registra uma mensagem de log ao criar o nó
        timer_ = this->create_wall_timer(               // cria um timer que chama timerCallback a cada segundo
            std::chrono::seconds(1),                    // intervalo de 1 segundo
            std::bind(&MyNode::timerCallback, this)     // vincula o método timerCallback ao timer
        );
    }

private:
    void timerCallback()
    {
        RCLCPP_INFO(this->get_logger(), "Hello %d", counter_); // registra o valor do contador e incrementa
        counter_++;
    }
    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
};

int main(int argc, char **argv)
{
    rclcpp:: init(argc, argv); // inicializa o rclcpp com os argumentos fornecidos
    auto node = std::make_shared<MyNode>(); // cria uma instância compartilhada do nó MyNode
    rclcpp::spin(node); // mantém o nó ativo para processar callbacks (até que se aperte Ctrl+C)
    rclcpp:: shutdown(); // finaliza o rclcpp
    return 0; // retorna 0 para indicar que o programa terminou com sucesso
}